src/Text/LLVM/PP.hs

```diff
@@ -1235,11 +1235,20 @@ ppDISubprogram' pp sp = "!DISubprogram"
 ppDISubprogram :: Fmt DISubprogram
 ppDISubprogram = ppDISubprogram' ppLabel
 
+ppDISubrange' :: Fmt i -> Fmt (DISubrange' i)
+ppDISubrange' pp sr =
+  let eimb pfx l mbr = either (Just . (pfx <+>) . l) (fmap ((pfx <+>) . mbr))
```

Comment:
It's really hard to grok what this code is doing at a glance, possibly because of all the pointfree code being smashed together. Perhaps adding an explicit type signature and/or a comment would help.

Reply:
I simplified this overall and pushed an update.  Let me know if you think more commenting/help is needed with the simpler version.

Reply:
It's not quite as intimidating as before, but I still think adding an explicit type signature would go a long way here.